pkg/cmd/server/api/validation/master.go

```diff
@@ -113,13 +113,22 @@ func ValidateMasterConfig(config *api.MasterConfig, fldPath *field.Path) Validat
 		if len(etcdConfigErrs.Errors) == 0 {
 			// Validate the etcdClientInfo with the internal etcdConfig
 			validationResults.AddErrors(ValidateEtcdConnectionInfo(config.EtcdClientInfo, config.EtcdConfig, fldPath.Child("etcdClientInfo"))...)
+
+			// Validate the kubernetesEtcdClientInfo with the internal etcdConfig
+			validationResults.AddErrors(ValidateEtcdConnectionInfo(*config.KubernetesMasterConfig.EtcdClientInfo, config.EtcdConfig, fldPath.Child("etcdClientInfo"))...)
```

Comment:
everything `KubernetesMasterConfig`-related should be validated in `ValidateKubernetesMasterConfig`, which ensures that `KubernetesMasterConfig` is not nil (which it could be here)